legion/legion/edi/server.py

```diff
@@ -208,9 +208,11 @@ def undeploy(model, version=None, grace_period=0, ignore_not_found=False):
 
     for model_service in model_services:
         if app.config['REGISTER_ON_GRAFANA']:
+            other_versions_exist = len(app.config['ENCLAVE'].get_models(model_service.id)) > 1
             LOGGER.info('Removing model\'s dashboard from Grafana (id={}, version={})'
                         .format(model_service.id, model_service.version))
-            if app.config['GRAFANA_CLIENT'].is_dashboard_exists(model_service.id, model_service.version):
+            if app.config['GRAFANA_CLIENT'].is_dashboard_exists(model_service.id, model_service.version) \
+                    and not other_versions_exist:
                 app.config['GRAFANA_CLIENT'].remove_dashboard_for_model(model_service.id,
                                                                         model_service.version)
             else:
```

Comment:
```python
LOGGER.info('Removing model\'s dashboard from Grafana has been skipped - \
                              dashboard does not exist')
```
The log is not entirely correct, what do you think about the following version?
```python
LOGGER.info('Removing model\'s dashboard from Grafana has been skipped')
```

Reply:
Agreed, this looks more accurate